pypika/queries.py

```diff
@@ -283,6 +300,24 @@ def make_columns(*names: Union[TypedTuple[str, str], str]) -> List[Column]:
     return columns
 
 
+class PeriodFor:
+    def __init__(self, name: str, start_column: Union[str, Column], end_column: Union[str, Column]) -> None:
+        self.name = name
+        self.start_column = start_column if isinstance(start_column, Column) else Column(start_column)
+        self.end_column = end_column if isinstance(end_column, Column) else Column(end_column)
+
+    def get_sql(self, **kwargs: Any) -> str:
+        quote_char = kwargs.get("quote_char")
+
+        period_for_sql = "PERIOD FOR {name} ({start_column_name},{end_column_name})".format(
```

Comment:
Is this syntax MSSQL only? 

MariaDB in this example wraps the start and end columns in a function https://mariadb.com/kb/en/temporal-data-tables/

Reply:
Hi @mikeengland 

I wondered about this, but it also exists in oracle and mssql. Oracle works as MariaDB, but for mssql the only valid period name is `SYSTEM_TIME`: e.g.

```sql
CREATE TABLE t
(
  id INT,
  valid_from DATETIME2,
  valid_to DATETIME2,
  PERIOD FOR SYSTEM_TIME (valid_from, valid_to)
) WITH SYSTEM VERSIONING
```

Reply:
I don't think it is an issue that needs to be addressed now, but it should be easy enough to create a custom create query builder where needed (like we do for Vertica) and override the period_for function so that it uses a PeriodFor class which generates SQL specific to that platform.

Reply:
Cool. Would you like me to make any more changes?

Reply:
No, it looks good. I'll merge. Thanks!

Reply:
Superb!

I'll do another PR for the docs